tutorials/dataframe/tdf012_using_jit.C

```diff
@@ -0,0 +1,63 @@
+/// \file
+/// \ingroup tutorial_tdataframe
+/// \notebook -nodraw
+///
+/// This tutorial illustrates how to save some typing when using TDataFrame
+/// by invoking functions that perform jit-compiling at runtime.
+///
+/// \macro_code
+///
+/// \date October 2017
+/// \author Guilherme Amadio
+
+#include "TRandom.h"
+#include "ROOT/TDataFrame.hxx"
+
+void tdf012_using_jit()
+{
+   // We will inefficiently calculate an approximation of pi by generating
+   // some data and doing very simple filtering and analysis on it
+
+   // We start by creating an empty dataframe where we will insert 10 million
+   // random points in a square of side 2.0 (that is, with an inscribed circle
+   // of radius 1.0)
+
+   size_t npoints = 10000000;
+   ROOT::Experimental::TDataFrame tdf(npoints);
+
+   // Make generators known to interpreter
+   gInterpreter->ProcessLine("TRandom rx, ry;");
+
+   // Use different seeds for independent streams
+   gInterpreter->ProcessLine("rx.SetSeed(1);");
+   gInterpreter->ProcessLine("ry.SetSeed(2);");
+
+   // Define what we want inside the dataframe. We do not need to define p as an array,
+   // but we do it here to demonstrate how to use jitting with TDataFrame
+
+   // NOTE: Although it's possible to use "for (auto&& x : p)" below, it will
+   // shadow the name of the data column "x", and may cause compilation failures
+   // if the local variable and the data column are of different types or the
+   // local x variable is declared in the global scope of the lambda function
```

Comment:
why do you have to use `x`?

Reply:
I just want to make the user aware of the potential problem, and when/why it happens.